src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs

```diff
@@ -884,32 +891,47 @@ private List<ArtifactSummary> harvestSummaries(List<string> paths)
             var origin = summary.Origin;
             if (string.IsNullOrEmpty(origin))
             {
-                throw Error.Argument($"Cannot load resource from summary. The {nameof(ArtifactSummary.Origin)} information is empty or missing.");
+                throw Error.Argument($"Unable to load resource from summary. The '{nameof(ArtifactSummary.Origin)}' information is unavailable.");
             }
 
             var pos = summary.Position;
             if (string.IsNullOrEmpty(pos))
             {
-                throw Error.Argument($"Cannot load resource from summary. The {nameof(ArtifactSummary.Position)} information is empty or missing.");
+                throw Error.Argument($"Unable to load resource from summary. The '{nameof(ArtifactSummary.Position)}' information is unavailable.");
             }
 
-            T result = null;
-            using (var navStream = DefaultNavigatorStreamFactory.Create(origin))
+            // Always use the current Xml/Json parser settings
+            var settings = _settings;
+            var factory = _navigatorFactory;
+            settings.XmlParserSettings.CopyTo(factory.XmlParsingSettings);
+            settings.JsonParserSettings.CopyTo(factory.JsonParsingSettings);
+
+            // Also use the current PoCo parser settings
+            var pocoSettings = new PocoBuilderSettings();
+            if (settings.ParserSettings != null)
             {
+                pocoSettings.AllowUnrecognizedEnums = settings.ParserSettings.AllowUnrecognizedEnums;
```

Comment:
Isn't this fragile?  What if we add new parser settings, these will then not get copied over?

Reply:
Marco suggests to implement members on settings classes themselves to copy internal state to another settings class. For example:

```
ParserSettings.CopyTo(PocoBuilderSettings settings) { /* ... */ }
PocoBuilderSettings.CopyTo(ParserSettings settings) { /* ... */ }
```

This is much better maintainable, esp. when adding new properties.

Reply:
I've created a new method for this:
`public void ParserSettings.CopyTo(PocoBuilderSettings settings)`